### ISSUE TITLE
Unable to include an image with empty alt attribute in Markdown

### ISSUE THREAD

**Issue by AishaBlake** [gatsbyjs/gatsby#20179]
## Description

In reviewing #19982, I realized I wasn't able to create an image with empty alt text. For that blog post, there was a case for it because Shannon was describing some of her images outside of alt text. This shouldn't happen often but I don't see a way to include an image that we want screen readers to skip entirely. If no alt text is provided, we use the name of the file.

When I tried `![](select-work-org-confusion.png)`, for example, I got something like `<img alt="select work org confusion"...` That behavior makes sense in most cases and is what happens in the browser when someone writing HTML forgets alt text entirely. I'm looking for something that gets me `<img alt=""...`

### Steps to reproduce

This occurred while reviewing the blog post linked above. In the Gatsby repo, you can:

- Run `cd www`.
- Run `gatsby develop`.
- Navigate to docs > blog > and then any blog post with images.
- Remove the alt text from an image and inspect it in the browser.

You can clone this [example blog](https://github.com/AishaBlake/mdx-alt-text-repro) and navigate to http://localhost:8000/hello-world/ to see a handful of different alt/title combinations. The code is in `content/blog/hello-world/index.md`.

### Expected result

There should be some option that results in an image with an empty alt attribute (and _not_ an image without alt attribute).

### Actual result

In every combination, both alt text and title are filled in somehow regardless of what information is provided.

### Environment

  System:
    OS: macOS Mojave 10.14.6
    CPU: (8) x64 Intel(R) Core(TM) i7-8569U CPU @ 2.80GHz
    Shell: 3.2.57 - /bin/bash
  Binaries:
    Node: 11.15.0 - ~/.nvm/versions/node/v11.15.0/bin/node
    Yarn: 1.19.2 - /usr/local/bin/yarn
    npm: 6.7.0 - ~/.nvm/versions/node/v11.15.0/bin/npm
  Languages:
    Python: 2.7.10 - /usr/bin/python
  Browsers:
    Chrome: 79.0.3945.88
    Safari: 13.0.4
  npmPackages:
    gatsby-design-tokens: ~1.0.10 => 1.0.10 
    gatsby-image: ^2.2.34 => 2.2.34 
    gatsby-plugin-canonical-urls: ^2.1.16 => 2.1.16 
    gatsby-plugin-catch-links: ^2.1.19 => 2.1.19 
    gatsby-plugin-emotion: ^4.1.16 => 4.1.16 
    gatsby-plugin-feed: ^2.3.23 => 2.3.23 
    gatsby-plugin-google-analytics: ^2.1.29 => 2.1.29 
    gatsby-plugin-google-tagmanager: ^2.1.18 => 2.1.18 
    gatsby-plugin-guess-js: ^1.1.26 => 1.1.26 
    gatsby-plugin-layout: ^1.1.16 => 1.1.16 
    gatsby-plugin-mailchimp: ^2.2.3 => 2.2.3 
    gatsby-plugin-manifest: ^2.2.31 => 2.2.31 
    gatsby-plugin-mdx: ^1.0.59 => 1.0.59 
    gatsby-plugin-netlify-cache: ^0.1.0 => 0.1.0 
    gatsby-plugin-nprogress: ^2.1.15 => 2.1.15 
    gatsby-plugin-offline: ^3.0.27 => 3.0.27 
    gatsby-plugin-react-helmet: ^3.1.16 => 3.1.16 
    gatsby-plugin-sharp: ^2.3.5 => 2.3.5 
    gatsby-plugin-sitemap: ^2.2.22 => 2.2.22 
    gatsby-plugin-theme-ui: ^0.2.43 => 0.2.43 
    gatsby-plugin-twitter: ^2.1.15 => 2.1.15 
    gatsby-plugin-typography: ^2.3.18 => 2.3.18 
    gatsby-remark-autolink-headers: ^2.1.19 => 2.1.19 
    gatsby-remark-code-titles: ^1.1.0 => 1.1.0 
    gatsby-remark-copy-linked-files: ^2.1.31 => 2.1.31 
    gatsby-remark-embed-video: ^1.7.1 => 1.7.1 
    gatsby-remark-graphviz: ^1.1.18 => 1.1.18 
    gatsby-remark-images: ^3.1.35 => 3.1.35 
    gatsby-remark-normalize-paths: ^1.0.0 => 1.0.0 
    gatsby-remark-prismjs: ^3.3.25 => 3.3.25 
    gatsby-remark-responsive-iframe: ^2.2.28 => 2.2.28 
    gatsby-remark-smartypants: ^2.1.17 => 2.1.17 
    gatsby-source-airtable: ^2.0.12 => 2.0.12 
    gatsby-source-filesystem: ^2.1.40 => 2.1.40 
    gatsby-source-npm-package-search: ^2.1.19 => 2.1.19 
    gatsby-transformer-csv: ^2.1.19 => 2.1.19 
    gatsby-transformer-documentationjs: ^4.1.20 => 4.1.20 
    gatsby-transformer-remark: ^2.6.39 => 2.6.39 
    gatsby-transformer-sharp: ^2.3.7 => 2.3.7 
    gatsby-transformer-yaml: ^2.2.18 => 2.2.18

**Comment by karthi07** [comment]
Can I work on this?

**Comment by AishaBlake** [comment]
As far as I'm concerned, you're welcome to work on this, @karthi07! It may still require some discussion, though. I'll add some additional context here and ping @johno in case he has anything to add.

In `gatsby-remark-image`, we [set a default value for alt text](https://github.com/gatsbyjs/gatsby/blob/master/packages/gatsby-remark-images/src/index.js#L173). One possible option we discussed for addressing this issue would be to add a keyword that could be passed in instead of alt text which would create an image element with an empty alt attribute. Something like `![SKIP](./some-img.png)`...

**Comment by AishaBlake** [comment]
@karthi07 please go ahead and run with this if you're interested!

**Comment by AishaBlake** [comment]
Hey, @karthi07! Checking in on this. Have you been able to work on this issue?

**Comment by AishaBlake** [comment]
@karthi07 I really appreciate your willingness to pick this up. Since it's been a couple of months since I've heard anything, I'm going to open this back up to the community. Please let me know if you've got the time to work on this!

**Comment by em-vo** [comment]
Hello @AishaBlake 
Do you have any update on this issue ? 
I wanted to add images with an empty alt in a blog post, and just discovered it was not possible.
I would be happy to work on this if needed

**Comment by tw15egan** [comment]
Also interested in having a fix for this, seems like we're running into this issue here: https://github.com/carbon-design-system/carbon-website/issues/1819

**Comment by motou** [comment]
I can take this issue since it should be helpful to generate an image with an empty alt attribute

**Comment by ronalde** [comment]
The fix dscribed above introduces a regression, especially from an accessibility perspective;  when declaring an empty alt-attribute using the keyword `GATSBY_EMPTY_ALT` *and* having set `showCaptions: true` in the options of the plugin in `gatsby-config.js`, the plugin produces unwanted HTML:

```html
<figure class="gatsby-resp-image-figure" ...>
    <span class="gatsby-resp-image-wrapper" style="...">
      <span class="gatsby-resp-image-background-image" style="..."></span>
  <img class="gatsby-resp-image-image" alt="" title="" src="/path.jpg" ...>
    </span>
    <figcaption class="gatsby-resp-image-figcaption"><p>GATSBY<em>EMPTY</em>ALT</p></figcaption>
  </figure>
<figcaption class="gatsby-resp-image-figcaption"><p>GATSBY<em>EMPTY</em>ALT</p></figcaption>
```
* the plugin produces the wanted (and fixed) empty value for the `alt`-attribute value of the `img`-element;
* but the plugin inserts (two?) unwanted  `figcaption`-elements, with unwanted non-sense content `<p>GATSBY<em>EMPTY</em>ALT</p>`.
 
I'm sorry I'm not experienced in Gatsby enough to try to fix it myself. I would like to urge the developers to-reopen this.

**Comment by ronalde** [comment]
Although this seems to cut it:
 
```diff
--- unpatched.index.js	2021-03-14 10:01:52.447293277 +0100
+++ patched.index.js	2021-03-14 09:28:20.860194553 +0100
@@ -117,7 +117,10 @@
                 var captionOptions = Array.isArray(options.showCaptions) ? options.showCaptions : options.showCaptions === true ? ["title", "alt"] : false;
 
                 if (captionOptions) {
-                  for (var _iterator = _createForOfIteratorHelperLoose(captionOptions), _step; !(_step = _iterator()).done;) {
+                    for (var _iterator = _createForOfIteratorHelperLoose(captionOptions), _step; !(_step = _iterator()).done;) {
+                        if (node.alt === EMPTY_ALT) {
+                            break
+                        }
                     var option = _step.value;
 
                     switch (option) {
```

`npm -l`  reports I'm using `gatsby-remark-images@3.11.1` but I was unable to find the right source here in github (I tried using the tagged version, ie https://raw.githubusercontent.com/gatsbyjs/gatsby/gatsby-remark-images%403.11.1/packages/gatsby-remark-images/src/index.js, but that differs a lot from my copy.

By looking at the current master it seems the problem is still there.

**Comment by AleksandrHovhannisyan** [comment]
Related: I wish I didn't have to type `GATSBY_EMPTY_ALT` in my MDX every time I wanted an empty alt attribute. Why can't I just do something like this in my MDX provider? `gatsby-remark-images` seems to always replaces `alt=""` with the name of the image (e.g., `puppy.png` => `alt="puppy"`), which is completely undesirable:

```tsx
const mdxComponents = {
  a: ({ children, ...otherProps }: UnderlinedLinkProps) => (
    <UnderlinedLink {...otherProps} initiallyUnderlined={true}>
      {children}
    </UnderlinedLink>
  ),
  // exclude title, intercept alt
  img: ({ style, alt, title, ...otherProps }: Omit<HTMLProps<HTMLImageElement>, 'crossOrigin'>) => {
    return (
      <img
        {...otherProps}
        // Why doesn't this work?
        alt={alt === '' ? 'GATSBY_EMPTY_ALT' : alt}
        style={{ ...style, maxWidth: '100%' }}
      />
    );
  },
}
```

If this worked, you could just use `alt=""` and replace it with whatever you want, in a single place. But that doesn't seem to work.
